app/src/main/java/org/simple/clinic/encounter/sync/EncounterSync.kt

```diff
@@ -41,9 +41,21 @@ class EncounterSync @Inject constructor(
       pushNetworkCall = { api.push(toRequest(it)) }
   )
 
-  private fun toRequest(encounters: List<ObservationsForEncounter>): EncounterPushRequest {
-    //TODO : Replace with an actual request
-    return EncounterPushRequest(listOf())
+  private fun toRequest(observationsForEncounter: List<ObservationsForEncounter>): EncounterPushRequest {
+    val encounterPayloads = observationsForEncounter.map { (encounter, bps) ->
+      val observationsPayload = EncounterObservationsPayload(bloodPressureMeasurements = bps.map { it.toPayload() })
```

Comment:
Correct me if I am wrong, but doesn't this replace the separate `BloodPressureSync`? At what point do we stop syncing blood pressure measurements separately?

We can discuss this offline.

Reply:
We are going to do this in the next PR.
Since this PR still doesn't create `Encounter` for new BPs, we have not removed `BloodPressureSync`. For the same reason we have not added `EncounterSync` to `DataSync` yet for frequently syncing.